deposit/src/test/java/edu/unc/lib/deposit/fcrepo4/StaffOnlyPermissionJobTest.java

```diff
@@ -0,0 +1,155 @@
+import edu.unc.lib.dl.persist.services.storage.StorageLocationTestHelper;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.CdrDeposit;
+import org.apache.commons.io.FileUtils;
+import org.apache.jena.rdf.model.Bag;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.vocabulary.RDF;
+import org.junit.Before;
+import org.junit.Test;
+
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositField;
+import edu.unc.lib.dl.fedora.PID;
+
+/**
+ * @author lfarrell
+ */
+public class StaffOnlyPermissionJobTest extends AbstractDepositJobTest {
+    private final static String LOC1_ID = "loc1";
+    private final static String FILE_CONTENT1 = "Some content";
+
+    private StaffOnlyPermissionJob job;
+    private Model model;
+    private StorageLocationManager locationManager;
+    private Bag depBag;
+    private StorageLocationTestHelper locTestHelper;
+    private File originalFile;
+
+    @Before
+    public void setup() throws Exception {
+        locTestHelper = new StorageLocationTestHelper();
+        locationManager = locTestHelper.createLocationManager(null);
+
+        job = new StaffOnlyPermissionJob();
+
+        job.setJobUUID(jobUUID);
+        job.setDepositUUID(depositUUID);
+        job.setDepositDirectory(depositDir);
+        setField(job, "locationManager", locationManager);
+        setField(job, "dataset", dataset);
+        setField(job, "depositsDirectory", depositsDirectory);
+        setField(job, "depositStatusFactory", depositStatusFactory);
+        setField(job, "jobStatusFactory", jobStatusFactory);
+        job.init();
+
+        model = job.getWritableModel();
+        depBag = model.createBag(depositPid.getRepositoryPath());
+        depBag.addProperty(Cdr.storageLocation, LOC1_ID);
+
+        originalFile = File.createTempFile("txt", "test");
+    }
+
+    @Test
+    public void testStaffOnly() throws Exception {
+        Map<String, String> depositStatus = new HashMap<>();
+        depositStatus.put(DepositField.staffOnly.name(), "true");
+        when(depositStatusFactory.get(eq(depositUUID))).thenReturn(depositStatus);
+
+        PID objPid = makePid();
+        Bag objBag = model.createBag(objPid.getRepositoryPath());
+        objBag.addProperty(RDF.type, Cdr.Work);
+
+        Resource fileResc = addFileObject(objBag, FILE_CONTENT1);
+        objBag.addProperty(Cdr.primaryObject, fileResc);
+        objBag.addProperty(RDF.type, Cdr.FileObject);
+
+        depBag.add(objBag);
+        job.closeModel();
+
+        job.run();
+
+        model = job.getReadOnlyModel();
+        Resource roles = model.getResource(objPid.getRepositoryPath());
+
+        assertTrue(roles.hasProperty(none.getProperty(), AUTHENTICATED_PRINC));
+        assertTrue(roles.hasProperty(none.getProperty(), PUBLIC_PRINC));
+
+        assertFalse(fileResc.hasProperty(none.getProperty()));
+
+        originalFile.deleteOnExit();
```

Comment:
You can do this immediately after you create the follow rather than in each test. Although i'll mention that again that the job doesn't do anything with files, so you don't really need to have actual files and staging locations set up

Reply:
No the badge branch isn't in here